cli/src/semdep/parsers/util.py

```diff
@@ -330,3 +333,38 @@ def lexeme(p: "Parser[A]") -> "Parser[A]":
     ),
 )
 json_doc = whitespace >> json_value
+
+
+# This is truly awful. Python's recursion limit is too small for some realistic package-lock.json files.
+# So we manually patch the parse and parse_partial methods of the json parser to increase it 😵‍💫
+
+
+class add_recursion_limit:
+    def __init__(self) -> None:
+        increase = environ.get("SEMGREP_PYTHON_RECURSION_LIMIT_INCREASE")
+        self.increase = int(increase) if increase else 500
+
+    def __enter__(self) -> None:
+        self.old_limit = getrecursionlimit()
+        setrecursionlimit(self.old_limit + self.increase)
+
+    def __exit__(self, type, value, tb):  # type: ignore
+        setrecursionlimit(self.old_limit)
```

Comment:
this would be simpler as a contextlib.contextmanager

https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager